Add exports to link a shared WebCore with WebKitLegacy for WinCairo

donny-dont opened the pull request:
#### 5e74bd17fb9dc8f9d55c9df54f384addae36e993
<pre>
Add exports to link a shared WebCore with WebKitLegacy for WinCairo
<a href="https://bugs.webkit.org/show_bug.cgi?id=248508">https://bugs.webkit.org/show_bug.cgi?id=248508</a>

Reviewed by Darin Adler.

Add `WEBCORE_EXPORT` everywhere needed to build a `SHARED` WebCore and
link with WebKitLegacy.

The `JSWindowProxy` class was exported completely. The `s_info` variable
is declared through `DECLARE_INFO` which doesn&apos;t export it. There is a
`DECLARE_EXPORT_INFO` macro but it uses `JS_EXPORT_PRIVATE` which is
incorrect in WebCore.

* Source/WebCore/accessibility/AccessibilityListBox.h:
* Source/WebCore/accessibility/win/AccessibilityObjectWrapperWin.h:
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/SimpleRange.cpp:
* Source/WebCore/dom/SimpleRange.h:
* Source/WebCore/editing/Editor.h:
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/VisibleUnits.h:
* Source/WebCore/html/HTMLFormElement.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DeprecatedGlobalSettings.h:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/FrameViewLayoutContext.h:
* Source/WebCore/page/win/FrameWin.h:
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/Pasteboard.h:
* Source/WebCore/platform/PlatformKeyboardEvent.h:
* Source/WebCore/platform/PlatformWheelEvent.h:
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Timer.h:
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/Font.h:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebCore/platform/graphics/cairo/RefPtrCairo.h:
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.h:
* Source/WebCore/platform/graphics/win/FullScreenController.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/curl/CurlCacheManager.h:
* Source/WebCore/platform/network/curl/CurlDownload.h:
* Source/WebCore/platform/network/curl/DownloadBundle.h:
* Source/WebCore/platform/win/BString.h:
* Source/WebCore/platform/win/PopupMenuWin.h:
* Source/WebCore/platform/win/SearchPopupMenuWin.h:
* Source/WebCore/platform/win/SystemInfo.h:
* Source/WebCore/platform/win/WebCoreBundleWin.h:
* Source/WebCore/platform/win/WebCoreTextRenderer.h:
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.h:

Canonical link: <a href="https://commits.webkit.org/257503@main">https://commits.webkit.org/257503@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97ff33001af4c090ba2d935734db4c58b6167546

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99030 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8235 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32180 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108432 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168682 "Failed to checkout and rebase branch from PR 6955") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102992 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8786 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85602 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91549 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106387 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104751 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6655 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90236 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33672 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88477 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21581 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76527 "Found 4 new API test failures: /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/network-error, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/console-api, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/js-exception, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/security-error (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2135 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23097 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2035 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45503 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5158 "'git push ...'") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/7001 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42572 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3452 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
Will the WinCairo EWS have this configuration? I think it’s critical that it does, otherwise this will quickly get broken.

donny-dont committed:
Add exports to link a shared WebCore with WebKitLegacy for WinCairo
https://bugs.webkit.org/show_bug.cgi?id=248508

Reviewed by Darin Adler.

Add `WEBCORE_EXPORT` everywhere needed to build a `SHARED` WebCore and
link with WebKitLegacy.

The `JSWindowProxy` class was exported completely. The `s_info` variable
is declared through `DECLARE_INFO` which doesn't export it. There is a
`DECLARE_EXPORT_INFO` macro but it uses `JS_EXPORT_PRIVATE` which is
incorrect in WebCore.

* Source/WebCore/accessibility/AccessibilityListBox.h:
* Source/WebCore/accessibility/win/AccessibilityObjectWrapperWin.h:
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/SimpleRange.cpp:
* Source/WebCore/dom/SimpleRange.h:
* Source/WebCore/editing/Editor.h:
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/VisibleUnits.h:
* Source/WebCore/html/HTMLFormElement.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DeprecatedGlobalSettings.h:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/FrameViewLayoutContext.h:
* Source/WebCore/page/win/FrameWin.h:
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/Pasteboard.h:
* Source/WebCore/platform/PlatformKeyboardEvent.h:
* Source/WebCore/platform/PlatformWheelEvent.h:
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Timer.h:
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/Font.h:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebCore/platform/graphics/cairo/RefPtrCairo.h:
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.h:
* Source/WebCore/platform/graphics/win/FullScreenController.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/curl/CurlCacheManager.h:
* Source/WebCore/platform/network/curl/CurlDownload.h:
* Source/WebCore/platform/network/curl/DownloadBundle.h:
* Source/WebCore/platform/win/BString.h:
* Source/WebCore/platform/win/PopupMenuWin.h:
* Source/WebCore/platform/win/SearchPopupMenuWin.h:
* Source/WebCore/platform/win/SystemInfo.h:
* Source/WebCore/platform/win/WebCoreBundleWin.h:
* Source/WebCore/platform/win/WebCoreTextRenderer.h:
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.h:

Canonical link: https://commits.webkit.org/257503@main
